### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by fed1k**
Hello from @Retky and @fed1k 
* Following database diagram for a clinic:
clinic_diagram
* Create a file named schema_based_on_diagram.sql where we implement the database from the diagram.
* Join tables from many-to-many relationships
* Add the FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    date_of_birth date
+);
+
+-- Create table named medical histories
+CREATE TABLE medical_histories(
+    id int primary key,
+    admitted_at timestamp,
+    patient_id int,
+    status varchar(255)
+);
```
Comment (KDlamini) [inline schema_based_on_diagram.sql:16]: -  Kindly reference the patients table using the patients_id as a foreign key in medical_histories.
For example:
```
FOREIGN KEY (patients_id) REFERENCES patients(id)
```

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+BEGIN;
+ALTER TABLE invoice_items
+    DROP COLUMN treatment_id,
+    ADD COLUMN treatment_id int REFERENCES treatments(id);
+COMMIT;
+
+
+
```
Comment (KDlamini) [inline schema_based_on_diagram.sql]: Even though the table is not shown on the diagram, please be sure to create a [joining table](https://dzone.com/articles/how-to-handle-a-many-to-many-relationship-in-datab#:~:text=A%20joining%20table%20is%20a,a%20much%20better%20data%20structure.) between medical_histories and treatments describing the many-to-many relationship they share.
For example:
```
-- Join Table
CREATE TABLE medical_histories_treatments (
...
)
```

- Remember to reference both medical_histories and treatments tables using foreign keys.
- Remember to create indexes for the new FOREIGN KEYS you added.

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+-- Create table named invoice_items
+CREATE TABLE invoice_items(
+    id int primary key,
+    unit_price decimal,
+    quantity int,
+    total_price decimal,
+    invoice_id int,
+    treatment_id int,
+    CONSTRAINT fkey_invoice_id FOREIGN KEY(invoice_id) REFERENCES invoices(id)
```
Comment (KDlamini) [inline schema_based_on_diagram.sql:39]: - Kindly add treatment id as a foreign key by referencing the treatments table as you did with invoice_id.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @fed1k, @Retky 🙋‍♀️,  

Great work so far! 💯 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (Retky) [reply schema_based_on_diagram.sql]: Thanks for the review @KDlamini
We'll implement the required changes 😃

**Review by rustam-tolipov — APPROVED** [review]
## Hi @fed1k 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**